CompuCell3D/core/Demos/PluginDemos/FocalPointPlasticityLinks/Simulation/FocalPointPlasticityLinksSteppables.py

```diff
@@ -0,0 +1,93 @@
+            print(f'neighbor_order: {link1j.getNeighborOrder()}, {fppd.neighborOrder}')
+            print(f'init_mcs: {link1j.getInitMCS()}, {fppd.initMCS}')
+
+            # Let's check the link's derived properties
+            print(f'length: {link1j.length}')
+            print(f'tension: {link1j.tension}')
+
+        # Now let's quickly get the list of all cells linked to cell 1
+        linked_list = self.get_fpp_linked_cells(cell1)
+        [print(f'Cell 1 has a link with cell {c.id}') for c in linked_list]
+
+        # Let's calculate how many cells cell1 is linked to
+        typej = {self.T1: self.T2, self.T2: self.T1}[cell1.type]
+        n1j = self.get_number_of_fpp_junctions_by_type(cell1, typej)
+        print(f'Cell {cell1.id} of type {cell1.type} has {n1j} junctions with type {typej}')
+
+        # While we're at it, let's see how many links there are
+        print(f'There are {self.get_number_of_fpp_links()} links')
+
+        # Every 100 steps, let's vary the target distance of all links
+        if mcs % 100 == 0:
+            for link in self.get_focal_point_plasticity_link_list():
+                max_distance = link.getMaxDistance()
+                target_distance = max_distance * random.random()
+                link.setMaxDistance(target_distance)
+
+        # Every 1000 steps, let's try to manually form type 1 - type 1 and type 2 - type 2 links
+        #   These links generally have a high activation energy, and so they won't form automatically
+        #   However, we can make them form when we create them by setting the activation energy for a particular
+        #   link to a permissible value
+        #   Our rule will be that each cell can only have with link with a cell of the same type
+        lambda_distance, target_distance, max_distance, activation_energy = 10, 5, 50, -50
+        if mcs % 1000 == 0:
+            for t in [self.T1, self.T2]:
+                cell_list = [cell for cell in self.cell_list_by_type(t)]
+                random.shuffle(cell_list)
+                celli, cellj = cell_list[0:2]
+                if self.get_number_of_fpp_junctions_by_type(celli, celli.type) == 0 and \
+                        self.get_number_of_fpp_junctions_by_type(cellj, cellj.type) == 0:
+                    new_link = self.new_fpp_link(celli, cellj, lambda_distance, target_distance, max_distance)
+                    new_link.setActivationEnergy(activation_energy)
+
+        # Every 1000 steps, let's randomly select a type 1 - type 2 link and break it
+        if mcs % 1000 == 0:
+            if self.get_number_of_fpp_links() > 0:
+                link_list = [link for link in self.get_focal_point_plasticity_link_list()]
+                random.shuffle(link_list)
+                self.delete_fpp_link(link_list[0])
+
+        # Every 10000 steps, let's break all links
+        if mcs % 10000 == 0:
+            [self.remove_all_cell_fpp_links(cell, links=True) for cell in self.cell_list]
```

Comment:
same here list comprehension without actully creating a list

Reply:
Just a concise way of removing all links. We could instead demo 

```python
for cell in self.cell_list:
    self.remove_all_cell_fpp_links(cell, links=True)
```